### ISSUE TITLE
How to check that output.bin file.

### ISSUE THREAD

**Issue by ektapadaliya** [ChrisAU/Dawg#2]
Hi,

I used this library in my app. I have one txt file in my app folder. How to use this library in app?

I am unable to find output.bin file in app?

Please help me.

Thanks,
Ekta

**Comment by cjnevin** [comment]
You have to create the bin file prior to using it in an app.

You can create a bin file using this line of code in a New **macOS -> Command Line Tool** project.

```
// Create a dawg binary file from a word list file.
assert(DawgBuilder.build(readingFrom: "~/input.txt", writingTo: "~/output.bin"))
```

Once that completes you should have a bin file in your home directory.

You can then load the bin file using:

```
let dawg = Dawg.load(from: "~/output.bin")
assert(dawg != nil)
```

This should only be done once ideally, as it's not a quick process, so you should create a singleton or inject it where you need to use the object.

In order to see if words exist you use the 'lookup' method.

```
// Returns false if the word is undefined
assert(dawg.lookup("plane"))
```

Please let me know if you need more information.

**Comment by ektapadaliya** [comment]
Hi Chris,

Thanks for your feedback.

Can you please tell me that how to open command line tool from xcode?

**Comment by ektapadaliya** [comment]
Can I use this library in single view application?

**Comment by cjnevin** [comment]
1. File -> New Project -> macOS -> Command Line Tool
2. Paste the line above.
3. Put your file in ~ and call it 'input.txt'
4. Press ⌘R
5. Wait until it completes.

**Comment by cjnevin** [comment]
Once you've compiled the .bin file you can use it wherever you like. It should just be compiled separately to a production app as it takes quite a while to process.

**Comment by ektapadaliya** [comment]
It gives me "assertion failed".

I have added "file.txt".

![screen shot 2017-01-30 at 4 04 13 pm](https://cloud.githubusercontent.com/assets/12406382/22419938/ca29a992-e705-11e6-963b-9220b9c1a8c9.png)

assert(DawgBuilder.build(readingFrom: "~/file.txt", writingTo: "~/output.bin"))

**Comment by cjnevin** [comment]
That is the incorrect path if you're going to bundle the file you'd need to find it using NSBundle. 

Just paste the input file in ~ directory and use the example provided it's easier.

**Comment by ektapadaliya** [comment]
Means I need to create one directory named "~".

**Comment by cjnevin** [comment]
On a mac/unix system `~` refers to the home directory.

**Comment by ektapadaliya** [comment]
Okay Now I got. Sorry. :(

**Comment by cjnevin** [comment]
Can I close this issue?

**Comment by ektapadaliya** [comment]
No. Please help me. I have added that file in root dir but still error.

**Comment by cjnevin** [comment]
You pasted it in ~ and called it input.txt
Then you loaded the file in the app by calling ~/input.txt?

**Comment by ektapadaliya** [comment]
Yes. in main.swift file 

import Foundation

print("Hello, World!")

// Create a dawg binary file from a word list file.
assert(DawgBuilder.build(readingFrom: "~/input.txt", writingTo: "~/output.bin"))

**Comment by ektapadaliya** [comment]
![screen shot 2017-01-30 at 4 22 49 pm](https://cloud.githubusercontent.com/assets/12406382/22420444/6686bd5a-e708-11e6-8f31-81ecc5e3e402.png)

Check this.

**Comment by cjnevin** [comment]
What format is your input.txt file?

It should contain each word on a new line

**Comment by ektapadaliya** [comment]
![screen shot 2017-01-30 at 4 26 07 pm](https://cloud.githubusercontent.com/assets/12406382/22420541/d70ed56c-e708-11e6-8b10-64e28842fe00.png)

**Comment by cjnevin** [comment]
If the build command fails it's because the file is not readable. 

Can you paste this above the 'build' method:

```
do {
let data = try String(contentsOfFile: "~/input.txt", encoding: String.Encoding.utf8)
} catch err {
print(err)
}

```

Then paste what you get?

Cheers
